lib/Applify.pm

```diff
@@ -270,11 +275,40 @@ sub _exit {
   exit $reason;
 }
 
+sub _generate_attribute_accessor {
+  my ($self, $args) = @_;
+  my $default = ref $args->{default} eq 'CODE' ? $args->{default} : sub { $args->{default} };
+  my $isa     = $args->{isa};
+  my $name    = $args->{name};
+
+  if (blessed $isa and $isa->can('check')) {
+    my $assert_method = $isa->has_coercion ? 'assert_coerce'    : 'assert_return';
+    my $prefix        = $isa->has_coercion ? 'Could not coerce' : 'Failed check';
+    return sub {
+      @_ == 1 && return exists $_[0]{$name} ? $_[0]{$name} : ($_[0]{$name} = $_[0]->$default);
+      die qq($prefix for "$name": $@) unless eval { $_[0]{$name} = $isa->$assert_method($_[1]); 1 };
```

Comment:
For `--people-ages` becoming `people_ages` in the error message, Applify already converts from perl variable names to the option names for output and accepting options from the command line, so is there a method we can use on `$name` to convert it in this error message?

Is something like:

```
$name =~ tr/_/-/;
die qq($prefix for "--$name": $@) ...
```

Too simple?

Reply:
Oh, it looks like `_attr_to_option` is what's used with `--` prepended by the caller?

Reply:
I cannot fix it here, since it will be very confusing if you do `$app->some_attribute($invalid_value)` later on, but I added a method called `_exit_invalid_argv()` which does the conversion.

Reply:
`_exit_invalid_argv()` uses steam to translate. Would a `local $self->{_instantiation} = 1` in `_app_new` with a suitable test here be better.

```perl
    my $assert_method = $isa->has_coercion ? 'assert_coerce'    : 'assert_return';
    my $prefix        = $isa->has_coercion ? 'Could not coerce' : 'Failed check';
    my $opt           = $self->_attr_to_option($name);
    return sub {
        @_ == 1 && return exists $_[0]{$name} ? $_[0]{$name} : ($_[0]{$name} = $_[0]->$default);
        die qq($prefix for "). $_[0]{_instantiation} ? "--$opt" : "$name" . qq(": $@)
            unless eval { $_[0]{$name} = $isa->$assert_method($_[1]); 1 };
    }
```

Reply:
I used `local $INSTANTIATING = 1`, but that's basically the same.